### DISCUSSION TITLE
Support hot reload

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [pyscript/pyscript#1822] [Proposals]
I would like to create an Panel playground similar to st-lite or gradio-lite. I have a proof of concept going at https://github.com/holoviz/panel/pull/5769 using Pyodide.

If py-script supported dynamically updating and rerunning the code inside `<py-script>` and the config inside `py-config` without having to reload pyodide and already loaded python packages then I believe I would like to use `pyscript` because that would make the `.html` file simpler for the users to inspect and for example embed in their web site.

I believe this feature would also help pyscript.com where I've also requested hot reload. Currently pyscript.com is not an efficient place to develop compared to stlite, gradio-lite or `panel convert` because its slow to update.

You can see the *hot reload* speed which I'm able to get in my proof of concept in the video below. Its amazing and even makes me want to skip developing on my laptop. Just move to browser because the hot reload is so much faster than server reload.

The window on the right is running Python and Panel in Pyodide. It updates everytime the code changes.

https://github.com/pyscript/pyscript/assets/42288570/b3af320d-5eb9-4f9c-bacc-66a5adbe22d4

**Comment by WebReflection** [comment]
this is possible with current RC by simply replacing the current node with another one identical but maybe with a different text content but I am not sure which version of PyScript you are running.

**Reply by MarcSkovMadsen** [reply]
The one Panel is pointing to. With Panel 1.3.0 it is 2023.03.1.

Are you saying this will be possible/ available in the next version of pyscript?

## Context

I've been playing around with pyodide in the weekends to create something like this.

I would like to load one instance of Pyodide across

- multiple `<py-script>` elements
- outside and inside iframes where the iframe is populated via `srcdoc` attribute.
- reloads as described above.

Especially reusing pyodide inside iframes seems not so easy as they execute in the `window` context. Not the `iframe.contextWindow` context. If if any code imports js it will get the outer `window` even if the python code is executed inside iframe.

I need to use iframes because Panel with templates renders into full `<html>` document with styles, js etc.

**Reply by WebReflection** [reply]
> Are you saying this will be possible/ available in the next version of pyscript?

yes because it bootstraps at runtime so you can hot-reload by replacing the live node and it's optional target with a clone of it and new content inside ... pyodide will be the exact same, no multiple bootstraps ever happen in PyScript next.
